datahub_sap_hana/ingestion.py

```diff
@@ -178,3 +205,182 @@ def _get_view_lineage_workunits(self) -> Iterable[MetadataWorkUnit]:
                 wu = item.as_workunit()
                 self.report.report_workunit(wu)
                 yield wu
+
+    def get_column_lineage_view_definitions(self, conn: Connection) -> Iterable[Tuple[str, str]]:
+
+        inspector = inspect(conn)
+        schema: List[Any] = inspector.get_schema_names()  # returns a list
+
+        for schema_name in schema:
+            views = inspector.get_view_names(
+                schema=schema_name)  # returns a list
+
+            for view_name in views:
+                view_sql = inspector.get_view_definition(
+                    view_name, schema_name)
+
+                if view_sql:
+                    yield (view_name, view_sql)
+
+    def _get_column_lineage_for_view(
+        self, view_name: str, view_sql: str
+    ) -> Tuple[str, List[Node]]:
+        """Extracts the columns and the sql definitions of a downstream view to build a lineage graph.
+        """
+
+        expression: DerivedTable = parse_one(view_sql)
+        selected_columns: List[str] = expression.named_selects
+
+        view_sql = view_sql.lower()
+        lineages = [
+            lineage(col_name.lower(), view_sql) for col_name in selected_columns
+        ]
+        return view_name, lineages
+
+    def parse_column_name(self, column_name: str):
+        """Removes the table/name and alias from the column name returned by sqlglot, eg: instead of 'hotel.room', it will return 'room'."""
+
+        parsed_name = column_name.split(".")
+        return parsed_name[-1]
+
+    def get_column_view_lineage_elements(
+        self, conn: Connection
+    ) -> Iterable[Tuple[Field, List[UpstreamField]]]:
+        """This function returns an iterable of tuples containing information about the lineage of columns in a view.
+        Each tuple contains a downstream field (a column in a view) and a list of upstream fields
+        (columns in other views or tables that are used to calculate/transform the downstream column).
+        """
+
+        for view_name, view_sql in self.get_column_lineage_view_definitions(conn):
+            view_name, column_lineages = self._get_column_lineage_for_view(
+                view_name, view_sql
+            )
+            # lineage_node represents the lineage of 1 column in sqlglot
+            # lineage_node.downstream is the datahub upstream
+            # each element of lineage_node.downstream is a node that represents a column in the source table
+
+            for lineage_node in column_lineages:
+                downstream_fields = Field(name=lineage_node.name)
+
+                upstream_fields_list: List[UpstreamField] = []
+
+                for column in lineage_node.downstream:
+                    upstream_fields = UpstreamField(
+                        name=self.parse_column_name(column.name),
+                        column=Column(name=column.source.name,
+                                      platform=self.platform),
```

Comment:
I was just looking at the docs and the `@platform_name` on [L102](https://github.com/contiamo/datahub-sap-hana/pull/46/files#diff-5876beb70b85912b2f979e0f4a646e791b621b78345936702e362b58e2ed9d20R102) actually gives us a couple of methods 

```
self.get_platform_name()
self.get_platform_id()
```

I think we should 

1. Change L102 to use `@platform_name("SAP Hana")`
2. use `self.get_platform_id()` here 
```suggestion
                                      platform=self.get_platform_id()),
```

Reply:
It is also worth noting, if you still want the platform id to be `hana` and not `sap-hana`, which is what it will generation, then you can override the id field

```python
def platform_name(
    platform_name: str, id: Optional[str] = None, doc_order: Optional[int] = None
)
```


meaning, you can use 

```python
@platform_name(platform_name="SAP Hana", id="hana")
```